packages/discovery/src/flatten/getASTIdentifiers.test.ts

```diff
@@ -0,0 +1,70 @@
+import { parse } from '@solidity-parser/parser'
+import { expect } from 'earl'
+
+import { getASTIdentifiers } from './getASTIdentifiers'
+
+const soliditySource = `pragma solidity ^0.8.0;
+
+import "Library.sol";
+
+library DefinedLibrary1 {
+    struct Buffer {
+        bytes data;
+    }
+
+    using DefinedLibrary1 for Buffer;
+
+    function push(Buffer storage buf, bytes memory data) internal {
+        UsedLibrary1.push(buf, data);
+    }
+}
+
+contract Contract1 {
+    event Log(EventLibrary.EventType emitIndex);
+    error Error1(ErrorLibrary.ErrorType errorCode);
+
+    struct Struct1 {
+        Lib.Buffer buffers;
+    }
+
+    function contract1Func(ParamLib1.Param param1, ParamLib2.Param param2) public returns (ReturnLib.ReturnType) {
+        if (heightArg > Library1.getHeight()) {
+            return heightArg;
+        } else {
+            return Library2.getSomeMagicValue();
+        }
+    }
+}`
+
+describe(getASTIdentifiers.name, () => {
+  it('gets unique identifiers', () => {
+    const ast = parse(soliditySource)
+    const idents = ast.children.flatMap((c) => getASTIdentifiers(c))
+    const result = new Set(idents)
+    expect([...result].sort()).toEqual(
+      [
+        'DefinedLibrary1',
+        'Buffer',
+        'bytes',
+        'data',
+        'buf',
+        'UsedLibrary1',
+        'Contract1',
+        'EventLibrary.EventType',
+        'emitIndex',
+        'ErrorLibrary.ErrorType',
+        'errorCode',
+        'Lib.Buffer',
+        'buffers',
+        'ParamLib1.Param',
+        'param1',
```

Comment:
I think that pulling identifiers this way can potentially be harmful via name clashes, or mostly not useful. What do we gain by having "param1" as identifier? Or "bytes"? We really care only about a few identifier types (contracts, libs, functions, imports). From this list we don't even know what each identifier is, meaning we need to have additional code somewhere else to prevent name clashes. And if we don't have it, then we can easily misinterpret a function call from param.